packages/aws-cdk-lib/custom-resource-config/test/custom-resource-config.test.ts

```diff
@@ -0,0 +1,547 @@
+import { Capture, Template, Match } from '../../assertions';
+import * as cdk from '../../core'
+import * as logs from '../../aws-logs';
+import * as s3 from '../../aws-s3';
+import * as ses from '../../aws-ses';
+import * as s3deploy from '../../aws-s3-deployment';
+import * as events from "../../aws-events";
+import * as targets from "../../aws-events-targets";
+import * as iam from "../../aws-iam";
+import { CustomResourceConfig } from '../custom-resource-config';
+
+describe('CustomResourceConfig empty retention', ()=> {
+    test('catch thrown error if no retention', ()=> {
+        const app = new cdk.App();
+        try {
+            CustomResourceConfig.of(app).addLogRetentionLifetime()
+        } catch (e) {
+            expect(e.message).toBe("Retention must have a value");
+        }
+    })
+})
+describe('aws-ses drop spams', () => {
```

Comment:
Name these according to the use-case of the feature, instead of according to implementation or testing details. 

One way to do it is a `describe` block for each of the three cases; one for when logRetention is set, one for when loggingConfig is set, and one for when both are undefined.